doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
+  \\ \\
+  For an ADT to be realized, we must have all arguments, so all these ports must be connected.
+  Now, if $L$ is $A$, then we have one free port, however if it is not, then this free port must not be free but instead be connected to the ADT above it.
+  \\ \\
+  For free variables which have no $\lambda$ to connect to, a symbol node is created, maintaining the invariant.
+  \\ \\
+  $\therefore$ AST→Net has one free port
+\end{Lemma}
+
+
+Now we need to prove that evaluating this net does not change this fact
+
+
+\begin{Theorem}{AST→Net→Eval has one free port}{AST->Net->Eval}
+  Let $A = Net(AST)$.
+  \\ \\
+  By lemma \ref{Lm:AST->Net} $A$ has one free port.
+  \\ \\
+  We must now show $Eval(A)$ does not change the number of free ports.
+  \\ \\
+  We can show this by simply considering how every reduction rule works. For the sake of brevity, we will only consider the three variations of rewrite rules present in the BOHM system WLOG.
+
+  \begin{tcolorbox}[title=Case 1 - Rewire, colframe=inner-box-1]
+    \begin{figure}[H]
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inetcell[]{a}{$g$}[5]
+          \inetcell[rotate = 180, below = 50 pt]{l}{\rotatebox[origin=c]{180}{$f$}}[5]
+          \inwire[]{a/1}{a}
+          \inwire[]{a/4}{d}
+          \outwire[]{l/1}{c}
+          \outwire[]{l/4}{b}
+          \cutWire{a.out}{l.out}{}{}
+        \end{tikzpicture}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.05\textwidth}
+        \huge{→}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \wirecross[]{a}{c}{b}{d}
+        \end{tikzpicture}
+      \end{subfigure}
+    \end{figure}
+
+    We can see that there is only one wire eliminated, the one connecting the main ports of node $f$ and node $g$.
+    \\ \\
+    The rest of the wires are simply rewired to each other, thus not effecting the total number of free ports
+  \end{tcolorbox}
+
+  \begin{tcolorbox}[title=Case 2 - Isolation, colframe=inner-box-1]
+    \begin{figure}[H]
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inetcell[]{a}{$f$}[6]
+          \inode[below = 40 pt]{b}{$g$}
+          \inwire[]{a/1}{a}
+          \inwire[]{a/3}{b}
+          \inwire[]{a/5}{c}
+          \cutWire{a.out}{b.north}{}{}
+        \end{tikzpicture}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.05\textwidth}
+        \huge{→}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inode[right = 20 pt]{b}{$⊗$}
+          \inwire[]{b}{b}
+          \draw (-1.5,-1.5) node{a}  .. controls (-1.5,1) and (0,-1.5) .. (0,1) node{c};
+        \end{tikzpicture}
+      \end{subfigure}
+    \end{figure}
+
+    Here we can see that like the first case the wire that connects the main ports are eliminated, and that $a$ and $c$ are rewired.
```

Comment:
This `f` node has three auxiliary ports, which isn't true for most of the current BOHM rules, and I don't think this pattern matches any of the rules either.

Are you thinking of erasure? Or of beta reduction?

Reply:
this case is very similar to what `If` does to `T` or `F`, and is the most general, as it rewrires a wire, and erases another branch.

the WLOG is used here, as all isolating cases act similarly

Reply:
We should mention that the number of ports varies, and be clear about what is under the "forall" in these WLOG cases.